internal/compiler/pathutils.rs

```diff
@@ -495,8 +496,8 @@ pub fn join(base: &Path, path: &Path) -> Option<PathBuf> {
         } else {
             path_str.to_string()
         };
-        let joined = clean_path_string(&format!("{base_str}{base_separator}{path_str}"));
-        Some(PathBuf::from(joined))
+        let joined = clean_path_string(&format_smolstr!("{base_str}{base_separator}{path_str}"));
+        Some(PathBuf::from(joined.to_string()))
```

Comment:
in this case, i *think* the use of smoltre might be counter-productive because Path uses a String behind the scene.

Reply:
I checked, and what I wrote is wrong - the conversion to a string is not a move as I expected, making this indeed a possible pessimization - that said overall this patch still is beneficial as indicated by the performance numbers and reductions in total allocations :thinking: